### PR TITLE
UNOMI-411 Bug fix in profile property condition and add getNestedProperty method

### PR DESCRIPTION
**Pull request by liatiusim**
When querying using "exist" operator in Condition, an existQuery is created and sent to Elasticsearch. According to ES documentation, exist return false the the field in null or empty list - "[]".

The case of an empty list wasn't handled in the eval() method of profile property condition evaluator. Need to return false if the field exist but has a value which is empty list.

### CODE REVIEW

File: api/src/main/java/org/apache/unomi/api/Profile.java [hunk 1]
```diff
@@ -94,6 +95,30 @@ public Object getProperty(String name) {
...
     }
 
+    /**
+     * Retrieves the value of the nested property identified by the specified name.
+     *
+     * @param name the name of the property to be retrieved, splited in the nested properties with "."
+     * @return the value of the property identified by the specified name
+     */
+    public Object getNestedProperty(String name) {
+        if (!name.contains(".")) {
```
Comment (sergehuber) [inline api/src/main/java/org/apache/unomi/api/Profile.java:105]: I'm wondering if there is a way we could avoid this code duplication but I don't know where to put it unless we create some kind of static class in the API.

**Review by sergehuber — CHANGES_REQUESTED** [review]
Thanks for the PR. I put a note about code duplication. If we could find a better way it would be preferable I think.

**Comment by sergehuber** [comment]
Could you also implement an integration test for this?

File: api/src/main/java/org/apache/unomi/api/Profile.java (hunk 1 above)
Comment (liatiusim) [reply api/src/main/java/org/apache/unomi/api/Profile.java:105]: Thank you for the comment. I'm not sure if I know where to put the common code, what do you think?

**Comment by liatiusim** [comment]
Added integration tests

File: api/src/main/java/org/apache/unomi/api/Profile.java (hunk 1 above)
Comment (sergehuber) [reply api/src/main/java/org/apache/unomi/api/Profile.java:105]: Yeah let's leave it like this for the time being but we'll probably have to refactor at some point.

**Review by sergehuber — APPROVED** [review]
Thanks for adding the IT test and for your patience. Looks good to me now I'll approve and merge.

**Pull request merged**